### ISSUE TITLE
[TW-404] Today and Filter Combining

### ISSUE THREAD

**Issue by taskwarrior** [GothenburgBitFactory/taskwarrior#439]
_John Warren on 2013-04-01T12:32:02Z says:_ 

 I'm quite novice to TaskWarrior so no doubt I've made a stupid mistake somewhere. I wanted to create a task today filter that would order all my "today" tasks by date/time. I started out with:
<pre>
report.today.description=Todays Jobs
report.today.columns=id,status,project,priority,due,start.active,due.countdown,tags,description
report.today.labels=ID,Status,Project,Pri,Due,A,Countdown,Tags,Description
report.today.sort=due+
report.today.filter=status.not:deleted due.before:tomorrow due.after:yesterday
</pre>
but this proved to be wrong so after reading a bit I changed it to:
<pre>
report.today.description=Todays Jobs
report.today.default=status:pending
report.today.columns=id,status,project,priority,due,start.active,due.countdown,tags,description
report.today.labels=ID,Status,Project,Pri,Due,A,Countdown,Tags,Description
report.today.sort=due+
report.today.filter=due.before:tomorrow status.not:deleted,completed status:pending or due.before:tomorrow status.not:deleted,completed status:waiting
</pre>
What struck me as odd was that I could not nest the or operation as a subset so (A or B) as a subset of the proceeding clause. Instead I get clause 1 and A or B which makes sense when reading but not what I was expecting.

So is it possible to have an (A or B) rather than having to restate things?
Why did results using tomorrow and yesterday miss out results that had been scheduled on midnight?
How can I get a scheduled component into this list so far I was thinking of adding:
<pre>
report.today.filter=scheduled.before:tomorrow status.not:deleted,completed status:pending or scheduled.before:tomorrow status.not:deleted,completed status:waiting
</pre>
Or should I be using all.before rather than due and scheduled?
Sorry for the noobieness of this post.

**Comment by taskwarrior** [comment]
Migrated metadata:
```
Created: 2013-04-01T12:32:02Z
Modified: 2014-02-09T01:42:32Z
```

**Comment by taskwarrior** [comment]
_Paul Beckingham on 2013-04-01T13:02:13Z says:_ 

 Hi John, welcome to Taskwarrior.

It looks like there are several issues going on at once here.  Let me address them individually:

1) This construct:
<pre>
status.not:deleted,completed
</pre>
is not supported.  You need one of the following:
<pre>
status.not:deleted status.not:completed
status.not:deleted and status.not:completed
</pre>

2) Your filter is wrong.  You have this:
<pre>
due.before:tomorrow status.not:deleted,completed status:pending or due.before:tomorrow status.not:deleted,completed status:waiting
</pre>
To simplify this, it is equivalent to:
<pre>
due:... AND status:... AND status:... OR due:... AND status:... AND status:...
</pre>
There is an implicit AND between filter clauses.  If you wish to include an OR then you need to consider operator precedence, and use parentheses.  Suggested reading:
http://taskwarrior.org/projects/taskwarrior/wiki/Feature_filters

3) Midnight is the default time when your daeformat does not include a time, and you don't specify one.  Suggested reading:
http://taskwarrior.org/projects/taskwarrior/wiki/Ck5
http://taskwarrior.org/projects/taskwarrior/wiki/Ck4

4) This would be better suited as a forum question, a support email, or an IRC discussion.  You're perfectly welcome to use any channel, but using the bug system sets off all kinds of alarms...

**Comment by taskwarrior** [comment]
_John Warren on 2013-04-01T13:32:17Z says:_ 

 Hi thanks for the quick reply and links to data sources

I agree order of filter terms is important I figured that out by trial and error.

My browser also implies I've read through all the pages you provided.

I also tried the IRC in the past but with limited success and have only just got my account activated.

I'll keep the reading - thanks for a great product and sorry for triggering alarms. I hope my other bug report can be so easily dealt with.
